src/main/java/com/angelbroking/smartapi/SmartConnect.java

```diff
@@ -800,10 +800,32 @@ public JSONObject logout() {
 			JSONObject response = smartAPIRequestHandler.postRequest(this.apiKey, url, params, accessToken);
 			return response;
 		} catch (Exception | SmartAPIException e) {
-			System.out.println(e.getMessage());
+			log.error(e.getMessage());
 			return null;
 		}
 	}
-    
+
+	/**
+	 * Get Individual Order Details
+	 *
+	 * @return JSONObject which contains order details from Smart API
+	 *
+	 */
+	public JSONObject getIndividualOrderDetails(String orderId) throws IOException, SmartAPIException {
+		try {
+			String url = routes.get("api.individual.order").concat(orderId);
+			return smartAPIRequestHandler.getRequest(this.apiKey, url, accessToken);
+		}catch (SmartAPIException ex) {
+			log.error("{} while getting individual order {}", SMART_API_EXCEPTION_OCCURRED, ex.toString());
+			throw new SmartAPIException(String.format("%s in getting individual order %s", SMART_API_EXCEPTION_ERROR_MSG, ex));
+		} catch (IOException ex) {
+			log.error("{} while getting individual order {}", IO_EXCEPTION_OCCURRED, ex.getMessage());

```

Comment:
Remove } catch (IOException ex) {
			log.error("{} while getting individual order {}", IO_EXCEPTION_OCCURRED, ex.getMessage());
			throw new IOException(String.format("%s in getting individual order %s", IO_EXCEPTION_ERROR_MSG, ex.getMessage()));
		} catch (JSONException ex) {
			log.error("{} while getting individual order {}", JSON_EXCEPTION_OCCURRED, ex.getMessage());
			throw new JSONException(String.format("%s in getting individual order %s", JSON_EXCEPTION_ERROR_MSG, ex.getMessage()));

		} 

They will never triggered.

Reply:
These unnessary catch blocks are now removed. Now only one catch block is added with SmartAPI exception is used to handled the exception.

` @Test
    public void testIndividualOrder_Success() throws SmartAPIException, IOException {
        String url = routes.get("api.individual.order");
        when(smartAPIRequestHandler.getRequest(eq(this.apiKey), eq(url), eq(this.accessToken))).thenReturn(createIndividualOrderResponse());
        try {
            JSONObject response = smartAPIRequestHandler.getRequest(this.apiKey, url, this.accessToken);
            JSONObject data = response.getJSONObject("data");
            assertNotNull(data);
        } catch (SmartAPIException ex) {
            log.error("{} while getting individual order {}", SMART_API_EXCEPTION_OCCURRED, ex.toString());
            throw new SmartAPIException(String.format("%s in getting individual order %s", SMART_API_EXCEPTION_ERROR_MSG, ex));
        }
    }`

Reply:
Actually I have test these catch blocks now and i found that these catch blocks effectively handled exceptions thrown by the **handle** method when invoking  APIs with okkhttp dependency. This ensured robust error handling during API calls. 
So i have again made the changes in the code and add that previous catch blocks.